lambdas/data-migration2/tests/test-granulesAndFiles.js

```diff
@@ -0,0 +1,538 @@
+  const {
+    knex,
+    testGranule,
+  } = t.context;
+
+  delete testGranule.collectionId;
+
+  await t.throwsAsync(
+    migrateGranuleRecord(testGranule, knex),
+    { name: 'SchemaValidationError' }
+  );
+});
+
+test.serial('migrateGranuleRecord handles nullable fields on source granule data', async (t) => {
+  const {
+    collectionCumulusId,
+    executionCumulusId,
+    granulePgModel,
+    knex,
+    testGranule,
+  } = t.context;
+
+  delete testGranule.pdrName;
+  delete testGranule.cmrLink;
+  delete testGranule.published;
+  delete testGranule.duration;
+  delete testGranule.files;
+  delete testGranule.error;
+  delete testGranule.productVolume;
+  delete testGranule.timeToPreprocess;
+  delete testGranule.beginningDateTime;
+  delete testGranule.endingDateTime;
+  delete testGranule.processingStartDateTime;
+  delete testGranule.processingEndDateTime;
+  delete testGranule.lastUpdateDateTime;
+  delete testGranule.timeToArchive;
+  delete testGranule.productionDateTime;
+  delete testGranule.timestamp;
+  delete testGranule.provider;
+
+  await migrateGranuleRecord(testGranule, knex);
+
+  const record = await granulePgModel.get(knex, {
+    granule_id: testGranule.granuleId,
+    collection_cumulus_id: collectionCumulusId,
+  });
+
+  t.deepEqual(
+    omit(record, ['updated_at', 'cumulus_id']),
+    {
+      granule_id: testGranule.granuleId,
+      status: testGranule.status,
+      collection_cumulus_id: collectionCumulusId,
+      published: testGranule.published,
+      duration: null,
+      time_to_archive: null,
+      time_to_process: null,
+      product_volume: null,
+      error: null,
+      cmr_link: null,
+      execution_cumulus_id: executionCumulusId,
+      pdr_cumulus_id: null,
+      provider_cumulus_id: null,
+      query_fields: null,
+      beginning_date_time: null,
+      ending_date_time: null,
+      last_update_date_time: null,
+      processing_end_date_time: null,
+      processing_start_date_time: null,
+      production_date_time: null,
+      timestamp: null,
+      created_at: new Date(testGranule.createdAt),
+    }
+  );
+});
+
+test.serial('migrateGranuleRecord throws RecordAlreadyMigrated error for already migrated record', async (t) => {
+  const {
+    knex,
+    testGranule,
+  } = t.context;
+
+  await migrateGranuleRecord(testGranule, knex);
+
+  await t.throwsAsync(
+    migrateGranuleRecord(testGranule, knex),
+    { instanceOf: RecordAlreadyMigrated }
+  );
+});
+
+test.serial('migrateFileRecord handles nullable fields on source file data', async (t) => {
+  const {
+    filePgModel,
+    knex,
+    testGranule,
+  } = t.context;
+
+  const testFile = testGranule.files[0];
+
+  delete testFile.bucket;
+  delete testFile.checksum;
+  delete testFile.checksumType;
+  delete testFile.fileName;
+  delete testFile.key;
+  delete testFile.path;
+  delete testFile.size;
+  delete testFile.source;
+
+  await migrateGranuleRecord(testGranule, knex);
```

Comment:
Again, somewhat inclined to just create the granule with the PG model and pass in that ID to `migrateFileRecord`

Reply:
[`2e2d579` (#2051)](https://github.com/nasa/cumulus/pull/2051/commits/2e2d579eabdb74b06e869b01d8e8993acdd98c9e)